skdatasets/keel.py

```diff
@@ -95,189 +103,309 @@ def _fetch(name, url, names, target_name, nfolds=5):
-            'yeast5': {'args': ['http://sci2s.ugr.es/keel/keel-dataset/datasets/imbalanced/imb_IRhigherThan9p1',
-                                ['Mcg', 'Gvh', 'Alm', 'Mit', 'Erl', 'Pox',
-                                 'Vac', 'Nuc', 'Class'],
-                                'Class']},
-            'yeast6': {'args': ['http://sci2s.ugr.es/keel/keel-dataset/datasets/imbalanced/imb_IRhigherThan9p1',
-                                ['Mcg', 'Gvh', 'Alm', 'Mit', 'Erl', 'Pox',
-                                 'Vac', 'Nuc', 'Class'],
-                                'Class']}}
+BASE_URL = 'http://sci2s.ugr.es/keel/'
+
+datasets = {
+    'abalone9-18': {
+        'args': [BASE_URL +
+                 'keel-dataset/datasets/imbalanced/imb_IRhigherThan9p1',
+                 ['Sex', 'Length', 'Diameter', 'Height',
+                  'Whole_weight', 'Shucked_weight',
+                  'Viscera_weight', 'Shell_weight',
+                  'Class'],
+                 'Class']
+        },
+    'abalone19': {
+        'args': [BASE_URL +
+                 'keel-dataset/datasets/imbalanced/imb_IRhigherThan9p1',
+                 ['Sex', 'Length', 'Diameter', 'Height',
+                  'Whole_weight', 'Shucked_weight',
+                  'Viscera_weight', 'Shell_weight',
+                  'Class'],
+                 'Class']
+        },
+    'balance': {
+        'args': [BASE_URL +
+                 'dataset/data/classification',
+                 ['left-weight', 'left-distance',
+                  'right-weight', 'right-distance', 'class'],
+                 'class']
+        },
+    'cleveland': {
+        'args': [BASE_URL +
+                 'dataset/data/classification',
+                 ['Age', 'Sex', 'Cp', 'Trestbps', 'Chol',
+                  'Fbs', 'Restecg', 'Thalach', 'Exang',
+                  'Oldpeak', 'Slope', 'Ca', 'Thal', 'Num'],
+                 'Num']
+        },
+    'cleveland-0_vs_4': {
+        'args': [BASE_URL +
+                 'keel-dataset/datasets/imbalanced/imb_IRhigherThan9p2',
+                 ['age', 'sex', 'cp', 'trestbps',
+                  'chol', 'fbs', 'restecg',
+                  'thalach', 'exang', 'oldpeak',
+                  'slope', 'ca', 'thal', 'num'],
+                 'num']
+        },
+    'ecoli': {
+        'args': [BASE_URL +
+                 'dataset/data/classification',
+                 ['mcg', 'gvh', 'lip', 'chg', 'aac', 'alm1',
+                  'alm2', 'class'],
+                 'class']
+        },
+    'ecoli-0-1-3-7_vs_2-6': {
+        'args': [BASE_URL +
+                 'keel-dataset/datasets/imbalanced/imb_IRhigherThan9p1',
+                 ['Mcg', 'Gvh', 'Lip', 'Chg', 'Aac',
+                  'Alm1', 'Alm2', 'Class'],
+                 'Class']
+        },
+    'ecoli-0-1-4-6_vs_5': {
+        'args': [BASE_URL +
+                 'keel-dataset/datasets/imbalanced/imb_IRhigherThan9p2',
+                 ['a1', 'a2', 'a3', 'a5', 'a6',
```

Comment:
Is there any reason for the missing 'a4' in some ecoli datasets?

Reply:
They are built that way: https://sci2s.ugr.es/keel/dataset.php?cod=981